statistics_service/native/src/impl/StatisticsService.cpp

```diff
@@ -1,164 +1,234 @@
-                            token >> myFloat;
-                            values.push_back(myFloat);
-                        }
-
-                        ++current_line_to_read;
-                        file.close();
+void StatisticsService::ReadTemperatures() {
+    std::ifstream file("/vendor/etc/temperatures.csv");
+
+    if (file.is_open()) {
+        ALOGI("%s:%d: file temperatures.csv open, reading values... ", __FUNCTION__, __LINE__);
+        std::string line;
+        
+        while(getline(file, line)) {
+            std::stringstream stream(line);
+            std::string value;
+            int current_value_index = 0;
+
+            while(getline(stream, value, ',')) {
+                switch (current_value_index) {
+                    case 0:
+                        all_cpu_temperatures_.push_back(std::stof(value));
+                        break;
+                    case 1:
+                        all_gpu_temperatures_.push_back(std::stof(value));
+                        break;
+                    case 2:
+                        all_ambient_temperatures_.push_back(std::stof(value));
+                        break;
+                    default:
+                        ALOGE("%s:%d: Error when pushing values", __FUNCTION__, __LINE__);
                         break;
-                    }
                 }
+                ++current_value_index;
             }
-            return values;
         }
+    } else {
+        ALOGE("%s:%d: std::cout << Error reading file ", __FUNCTION__, __LINE__);
+    }
+}
 
-        float StatisticsService::getValue(CallbacksId id)
-        {
-            std::vector<float> read_values = readTemperatures();
-
-            switch (id) {
-                CallbacksId::CpuTemperature:
-                    return read_values[0];
-                CallbacksId::GpuTemperature:
-                    return read_values[1];
-                CallbacksId::AmbientTemperature:
-                    return read_values[2];
-                CallbacksId::AverageCpuTemperature:
-                    return read_values[0];
-                CallbacksId::AverageGpuTemperature:
-                    return read_values[1];
-                CallbacksId::AverageAmbientTemperature:
-                    return read_values[2];
-                CallbacksId::MaxCpuTemperature:
-                    return read_values[0];
-                CallbacksId::MaxGpuTemperature:
-                    return read_values[1];
-                CallbacksId::MaxAmbientTemperature:
-                    return read_values[2];
-                default:
-                    return 0.0;
-            }
-        }
+float StatisticsService::GetCpuValue() {
+    float cpu_value = *(all_cpu_temperatures_.begin() + iterator_all_cpu_temps_);
+    ++iterator_all_cpu_temps_;
+    iterated_cpu_values_.push_back(cpu_value);
+
+    if(iterator_all_cpu_temps_ == all_cpu_temperatures_.size())
+        iterator_all_cpu_temps_ = 0;
+    
+    return cpu_value;

```

Comment:
idiomatic way pentru a implemneta aceasta functie e:
```suggestion
    return all_cpu_temperatures[++iteator_all_cpu_temps_%all_cpu_temperatores.size()];

```

Reply:
Ai putea sa-mi explici de ce anume faci impartirea aceea cu rest?
Si alta intrebare ar fi, daca fac asa ar trebui sa fac altcumva verificarea si sa resetez "iterator_all_cpu_temps_" sau ar trebui sa scap de tot  de restul codului si sa raman doar cu acel return sugerat de tine? Si totodata ma gandesc ce sa fac cu linia asta de cod "iterated_cpu_values_.push_back(cpu_value);", daca las doar return-ul

Reply:
operaturl "%" calculeaza restul impartirii indexului tau(iterator_all_cpu_tems) la numarul de termperaturi stocate(all_cpu_temperatures.size()). Acest rest este garantat sa fie in intervalul [0, all_cpu_temperatures.size() - 1]. In felul acesta, nu trebuie sa resetezi valoarea lui iterator_all_cpu_temps cand trece de all_cpu_temeratures.size().
Legat de overflow, acesta este defined behavior pentru unsigned types, si in momentul cand vei incrementa peste valoarea maxima, variabila va face "wrap-around" la 0.(NOTE: overflow este undefined behavior pentru signed integer types).